### PR TITLE
Add is_requester config setting.

### PR DESCRIPTION
**Pull request by jyao1**
Fix: https://github.com/DMTF/libspdm/issues/1523

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

spdm_emu fix: https://github.com/DMTF/spdm-emu/pull/192

### CODE REVIEW

**Comment by jyao1** [comment]
> * Update documentation to say that this is required.

I have concern to make it required in normal case. That will become a incompatible change.

**Comment by steven-bellock** [comment]
> > * Update documentation to say that this is required.
> 
> I have concern to make it required in normal case. That will become a incompatible change.

Ok we can save it for 3.0.

File: include/library/spdm_common_lib.h [hunk 1]
```diff
@@ -95,6 +95,14 @@ typedef enum {
 
     /* My provisioned certificate slot_id (for slot_id - 0xFF, default 0)*/
     LIBSPDM_DATA_LOCAL_PUBLIC_CERT_CHAIN_DEFAULT_SLOT_ID,
+
+    /* if the context is for a requester. It only needs to be set in VCA cache.
+     * In normal flow, the value is set in GET_VERSION or VERSION automatically.
+     * 0x0 means responder
```
Comment (steven-bellock) [inline include/library/spdm_common_lib.h:101]: `false means Responder`
`true means Requester`

**Comment by jyao1** [comment]
> Ok we can save it for 3.0.

Agree.

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 2]
```diff
@@ -465,6 +465,12 @@ libspdm_return_t libspdm_set_data(void *context, libspdm_data_type_t data_type,
                          sizeof(spdm_context->transcript.message_a.buffer),
                          data, data_size);
         break;
+    case LIBSPDM_DATA_IS_REQUESTER:
+        if (data_size != sizeof(uint8_t)) {
```
[the remark this answers is not in the document]
Comment (jyao1) [reply library/spdm_common_lib/libspdm_com_context_data.c]: Fixed

File: include/library/spdm_common_lib.h (hunk 1 above)
Comment (jyao1) [reply include/library/spdm_common_lib.h]: Fixed

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**